Makefile

```diff
@@ -16,3 +16,10 @@ go/mod:
 	cd godeltaprof/ && GO111MODULE=on go mod tidy
 	cd x/k6/ && GO111MODULE=on go mod download
 	cd x/k6/ && GO111MODULE=on go mod tidy
+
+# https://github.com/grafana/pyroscope-go/issues/129
+.PHONY: gotip/fix
+gotip/fix:
```

Comment:
Nit: Maybe `gotip/check` since if I understand it properly, this is just checking compatibility issues?

Reply:
It is not just checking, it is modifying go.mod by upgrading the golang.org/x/tools depdendency

The  toolchain check is my attempt to make sure no toolchain directives are added during upgrade. TBH I dont know why, but when I upgrade the dependency with go 1.23 I get a toolchain directive

```diff
--- a/godeltaprof/compat/go.mod
+++ b/godeltaprof/compat/go.mod
@@ -1,17 +1,19 @@
 module github.com/grafana/pyroscope-go/godeltaprof/compat
 
-go 1.18
+go 1.22.0
+
+toolchain go1.23.0
```
However if I upgrade with gotip, no toolchain added/